src/core.jl

```diff
@@ -150,7 +153,7 @@ struct OperationContext
     control_ops::Vector{Vector{Any}}  # Can't make Operation to break type cycle
     names::Vector{String}
     while_context::Vector{tensorflow.WhileContextDef}
-    devices::Vector{Device}
+    devices::Vector{AbstractDevice}
     is_top_level::Ref{Bool}
 end
```

Comment:
We should fix this type, to be parametric.

Reply:
Might as well just reduce complexity and turn that back to `Device` now that there's only device type again.